demes/src/specification.rs

```diff
@@ -1059,6 +1059,42 @@ impl Deme {
     pub fn proportions(&self) -> &[Proportion] {
         &self.proportions
     }
+
+    /// Size of Deme at a given time
+    pub fn size_at<F: Into<f64>>(&self, time: F) -> Result<Option<DemeSize>, DemesError> {
+        let time: f64 = time.into();
+        Time::from(time).validate(DemesError::DemeError)?;
+
+        if time == f64::INFINITY && self.start_time == f64::INFINITY {
+            return Ok(Some(self.epochs()[0].start_size));
+        };
+
+        let epoch = self.epochs().iter().find(|x| {
+            x.time_interval()
+                .contains_exclusive_start_inclusive_end(time)
+        });
+
+        if epoch == None {
+            // no epoch contains time
+            return Ok(None);
+        };
+        let epoch = epoch.unwrap();
+
+        let start_time = f64::from(epoch.start_time);
+        let end_time = f64::from(epoch.end_time);
+        let time_span = start_time - end_time;
+        let dt = (start_time - time) / time_span;
+        let size = match epoch.size_function() {
+            SizeFunction::Constant => epoch.end_size,
+            SizeFunction::Linear => DemeSize::from(f64::from(epoch.start_size) + time_span * dt),
+            SizeFunction::Exponential => {
+                let r = (f64::from(epoch.end_size) / f64::from(epoch.start_size)).log10();
```

Comment:
I think we have a problem in here -- aren't we mixing bases of exponents (log10 and then exp later).  And the growth rate, `r` should involve the epoch duration: (ln(end_size) - ln(start_size))/epoch_duration.

In the test, the code:

```rust
    println!("{:?}", g.deme(0).size_at(1e-5).unwrap());
```

will print out 402, which seems off given the final size of 1000.

Reply:
Aha -- I see what you did with "dt" there. The mixing of bases here is a problem, and replacing `.log10` with `.ln` gives the correct size for t = 1e-5 but cases other tests to fail.

Reply:
> correct size for t = 1e-5 but cases other tests to fail.

but they pass if your replace log10 with ln in the test.